lib/Fhp/FinTs.php

```diff
@@ -589,6 +589,8 @@ public function selectTanMode($tanMode, $tanMedium = null)
         }
         $this->selectedTanMode = $tanMode instanceof TanMode ? $tanMode->getId() : $tanMode;
         $this->selectedTanMedium = $tanMedium instanceof TanMedium ? $tanMedium->getName() : $tanMedium;
+		
+		$this->kundensystemId = $this->options->kundensystemId;
```

Comment:
Why does this happen within `selectTanMode()` and not in the same place where the `persist()` stuff is read back?

Reply:
Because it works here, no other reason :woman_shrugging:

Reply:
I'd move it into the `new()` function around line 88, and either put it into the `else` branch (i.e. only run it when `$persistedInstance === null`) or into an `if ($this->options->kundensystemId === null)`.

(Without that `if` protection, it will break any user that doesn't populate the field, because it will overwrite the restored kundensystemId with `null` -- in fact I'm surprised no tests are breaking.)